internal/k8s/statusupdater.go

```diff
@@ -152,21 +153,46 @@ func (suh *StatusUpdateHandler) Writer() StatusUpdater {
 // StatusUpdater describes an interface to send status updates somewhere.
 type StatusUpdater interface {
 	Update(name, namespace string, gvr schema.GroupVersionResource, mutator StatusMutator)
+	SendStatusUpdate(su StatusUpdate)
```

Comment:
We don't need to add an interface method here. If you want the convenience, then we can add a free function:

```Go
func SendStatusUpdate(u StatusUpdater, s StatusUpdate) {
        u.Update(s.Name.Name, s.Name.Namespace, s.Resource, s.Mutator)
}
```